server/src/connectionHandlers/onDefinition.ts

```diff
@@ -68,6 +68,17 @@ export function onDefinitionHandler (textDocumentPositionParams: TextDocumentPos
         })
       }
 
+      const exactSymbol = analyzer.getGlobalDeclarationSymbols(documentUri).find((symbol) => symbol.name === word && analyzer.positionIsInRange(position.line, position.character, symbol.location.range))
+
+      if (exactSymbol?.history !== undefined) {
+        exactSymbol.history.forEach((location) => {
+          definitions.push({
```

Comment:
You should replace the old way of getting results with this one which is accurate rather than concatenating the two. Should also get rid of duplicates.

Reply:
yes, if there is a difference in range it will result in another definition. As long as the URI and the range are the same, there won't be duplicates.

One thing to note is that I create the locations assuming the variable starts at index 0. If somehow users put a space or indentation before the variable. The highlight could be imprecise. I could do the `analyzer.analyze()` on every URI found in the operations to get the correct variable range but it costs performances. 
![image](https://github.com/yoctoproject/vscode-bitbake/assets/47988425/34ca3fa0-1dfe-41c4-835b-d10e1fc8462b)